tox.ini

```diff
@@ -66,6 +66,13 @@ commands =
         -m pytest --ignore={[vars]tst_path}integration -v --tb native -s {posargs}
     coverage report
 
+[testenv:shellcheck]
+description = Run shellcheck tests
+deps =
+    shellcheck-py==0.9.0.2
+whitelist-externals = /bin/bash
+commands = /bin/bash -c 'shellcheck --color=always $(git ls-files "*.bash" "*.sh")'
```

Comment:
Bash call is necessary here to execute and expand `git ls-files...` subshell.

Reply:
Moved to allowlist_externals in https://github.com/canonical/mysql-k8s-operator/pull/131/commits/a42e179b638397d4e8facee79d380acaf0faca66 as recommended by tox.

Reply:
> Right now, it's also ignoring untracked files which haven't yet been added to git

Yes, to avoid long blacklisting for folders like `.tox/` when using plain `find . -name "*.sh"` call.
For CI/CD it doesn't matter as all files delivered there through Git. Is it a problem?

Reply:
> but it also might be run locally

Yes. My rule of the thumb is: run "tox -e lint" for the ready commits before "git push ...".
Shellcheck is very picky and all local files will produce tons of noises... let's try step-by-step here.